README.md

```diff
@@ -106,6 +120,22 @@ aggregators:
     hosts : "agg-31"
     group : l3-agg
 
+samplers:
+  - names       : *sampler-hosts
+    hosts       : *sampler-hosts
+    ports       : *sampler-ports
```

Comment:
I think this is part of the host spec created in the 'hosts' section.

Reply:
It is. This is how maestro specifies the particular ldmsd on which to load samplers, the idea being that not all computes on a cluster will be running the exact same sampler plugins. I also use the sampler's "names" in the configuration to generate each sampler's instance names.

Reply:
But each host in the 'hosts' list can be used to look up the hostname and the port. It doesn't need to be replicated here.

And the 'names' section can be just a single value. As it is it creates a separate configuration object for each host, there only needs to be one configuration object and maestro can use that to generate the configuration for each sampler.

That said, I think we need some capability to set host-specific values, e.g.
```
load name=meminfo
config name=meminfo instance=%hostname/meminfo producer=%hostname job_set=%hostname/slurm
```

Doing it this way, we can significantly reduce the amount of data that's sitting in the etcd key-value store.

Reply:
BTW, note that the 'name' in the hosts section is not the hostname, it's the name of the host-entry that encapsulates both the host name and the listening port.

Reply:
I actually think that is confusing and we should probably change it to make it more obvious. For example change the 'host' entry in the 'hosts' section to be 'addr' or something like that. And then 'addr' can either be the hostname, e.g. 'orion-01', or the ip address '10.10.0.71'

Reply:
OK. I've removed host/ports from the configuration example. I actually wrote the back end to grab the host/port information from the 'hosts' section, so this technically already functions how you'd like on the back end. As far as using "addr" instead of "hosts" in the hosts section. Does this mean you'd like to change the aggregator configuration to use "addr" as well? I think we should be thorough with the adoption of whatever keyword we decide on so we don't further confuse people.

Host-specific values are currently handled by allowing null values for attributes. The values are all parsed before being submitted to a ldmsd, and if a value is None, the attribute isn't included in the configuration string.